src/http.rs

```diff
@@ -406,6 +432,39 @@ where
                 path: path.into(),
                 authenticated,
                 encrypted,
+                extension: false,
+            })
+            .unwrap(),
+        )
+        .send_and_await_response(5)
+        .unwrap();
+    let Ok(Message::Response { body, .. }) = res else {
+        return Err(HttpServerError::PathBindError {
+            error: "http_server timed out".to_string(),
+        });
+    };
+    let Ok(resp) = serde_json::from_slice::<std::result::Result<(), HttpServerError>>(&body) else {
+        return Err(HttpServerError::PathBindError {
+            error: "http_server gave unexpected response".to_string(),
+        });
+    };
+    resp
+}
+
+/// Register a WebSockets path with the HTTP server specifically for sending and
+/// receiving system messages from a runtime extension. Only use this if you are
+/// writing a runtime extension.
```

Comment:
This looks good but I'd change the comment slightly, since the WS for the ext can also be used to serve a FE: its both.